packages/rich-text/src/plugins/Hyperlink/index.tsx

```diff
@@ -1,316 +1,2 @@
-import { isLinkActive, unwrapLink } from '../../helpers/editor';
-import { transformRemove } from '../../helpers/transformers';
-import { useSdkContext } from '../../SdkProvider';
-import {
-  RichTextPlugin,
-  CustomRenderElementProps,
-  CustomElement,
-  RichTextEditor,
-} from '../../types';
-import { withLinkTracking } from '../links-tracking';
-import { ToolbarButton } from '../shared/ToolbarButton';
-import { EntryAssetTooltip } from './EntryAssetTooltip';
-import { addOrEditLink } from './HyperlinkModal';
-import { hasText } from './utils';
-
-const styles = {
-  hyperlinkWrapper: css({
-    display: 'inline',
-    position: 'static',
-    a: {
-      fontSize: 'inherit !important',
-    },
-  }),
-  hyperlink: css({
-    fontSize: 'inherit !important',
-    display: 'inline !important',
-    '&:hover': {
-      fill: tokens.gray900,
-    },
-    '&:focus': {
-      fill: tokens.gray900,
-    },
-  }),
-  hyperlinkIEFallback: css({
-    color: '#1683d0',
-    textDecoration: 'underline',
-  }),
-  // TODO: use these styles once we can use the icon
-  hyperlinkIcon: css({
-    position: 'relative',
-    top: '4px',
-    height: '14px',
-    margin: '0 -2px 0 -1px',
-    webkitTransition: 'fill 100ms ease-in-out',
-    transition: 'fill 100ms ease-in-out',
-    '&:hover': {
-      fill: tokens.gray900,
-    },
-    '&:focus': {
-      fill: tokens.gray900,
-    },
-  }),
-};
-
-type HyperlinkElementProps = CustomRenderElementProps<{
-  uri?: string;
-  target?: Link;
-  onEntityFetchComplete?: VoidFunction;
-}>;
-
-function UrlHyperlink(props: HyperlinkElementProps) {
-  const editor = useContentfulEditor();
-  const isReadOnly = useReadOnly();
-  const sdk: FieldExtensionSDK = useSdkContext();
-  const { uri } = props.element.data;
-
-  function handleClick(event: React.MouseEvent<HTMLAnchorElement>) {
-    event.preventDefault();
-    event.stopPropagation();
-    if (!editor) return;
-    addOrEditLink(editor, sdk, editor.tracking.onViewportAction);
-  }
-
-  return (
-    <Tooltip
-      content={uri}
-      targetWrapperClassName={styles.hyperlinkWrapper}
-      placement="bottom"
-      maxWidth="auto">
-      <TextLink
-        as="a"
-        href={uri}
-        rel="noopener noreferrer"
-        onClick={handleClick}
-        isDisabled={isReadOnly}
-        className={styles.hyperlink}>
-        {props.children}
-      </TextLink>
-    </Tooltip>
-  );
-}
-
-function EntityHyperlink(props: HyperlinkElementProps) {
-  const editor = useContentfulEditor();
-  const isReadOnly = useReadOnly();
-  const sdk: FieldExtensionSDK = useSdkContext();
-  const { target } = props.element.data;
-  const { onEntityFetchComplete } = props;
-
-  React.useEffect(() => {
-    // The real entity loading happens in the tooltip
-    // Since that is deferred the link is considered rendered as soon
-    // the component mounts (link text displayed)
-    onEntityFetchComplete?.();
-  }, [onEntityFetchComplete]);
```

Comment:
The `onEntityFetchComplete` will now be called once the actual fetch is complete in the `useEntityInfo` hook. However, I wonder if we should be calling anything at all. I checked master and it seems we don't have the equivalent of `onEntityFetchComplete` for hyperlinks? cc @IBratWurst

Reply:
I think that was a bug that we fixed in the new RTE compared to the old one.

All node types relying on an entry/asset should report once they received their entity (i.e. it got fetched)

Reply:
But the old editor [doesn't report on entity hyperlinks](https://github.com/contentful/field-editors/search?q=linkRendered), only embeds?